### PR TITLE
[CMAKE] Avoid the use of LOCATION properties

### PR DESCRIPTION
**Pull request by ThFabba**
This should get modern CMake versions to the same speed as our ancient RosBE version.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ endif()
 
 if(POLICY CMP0026)
     # Allow use of the LOCATION property
-    cmake_policy(SET CMP0026 OLD)
+    cmake_policy(SET CMP0026 NEW)
```
Comment (AmineKhaldi) [inline CMakeLists.txt]: NEW is default so you may omit this block now.

File: boot/CMakeLists.txt [hunk 2]
```diff
@@ -59,8 +59,8 @@ set(ISO_VOLNAME      "ReactOS")             # For both the Volume ID and the Vol
 
 ## BootCD
 # Create the file list
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootcd.lst "")
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootcd.lst "${CMAKE_CURRENT_BINARY_DIR}/empty\n")
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootcd.cmake.lst "")
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootcd.cmake.lst "${CMAKE_CURRENT_BINARY_DIR}/empty\n")
```
Comment (HBelusca) [inline boot/CMakeLists.txt]: Question: having the ".cmake." appended in the file name means that the file will contain also cmake commands in it?

File: CMakeLists.txt (hunk 1 above)
Comment (ThFabba) [reply CMakeLists.txt]: Actually if I read the documentation right, the default is always OLD unless you have a `cmake_policy(VERSION major.minor[.patch[.tweak]])` command?

File: boot/CMakeLists.txt (hunk 2 above)
Comment (ThFabba) [reply boot/CMakeLists.txt]: We've talked about this on IRC, but I'll say it again here so it's on the record: the file will contain generator expressions like $<TARGET_FILE:some_module>. So it needs to be processed by CMake, but doesn't contain CMake commands.

**Pull request merged**